src/lib.rs

```diff
@@ -93,10 +94,40 @@ lazy_static! {
     pub static ref PERFORMANCE_DATA: Mutex<PerformanceData> = Mutex::new(PerformanceData::new());
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct AperfStat {
+    pub time: TimeEnum,
+    pub name: String,
+    pub data: HashMap<String, u64>,
+}
+
+impl AperfStat {
+    fn new(name: String) -> Self {
+        AperfStat {
+            time: TimeEnum::DateTime(Utc::now()),
+            name,
+            data: HashMap::new(),
+        }
+    }
+
+    fn measure<F>(&mut self, name: String, func: F, datatype: &mut data::DataType) -> Result<()>
+    where
+        F: Fn(&mut data::DataType) -> Result<()>,
+    {
+        let start_time = time::Instant::now();
+        func(datatype)?;
```

Comment:
Could you not just have this be generic, taking a no-args function and executing it?

That should be possible. The caller creates a lambda that captures everything.

Reply:
Yeah. Changing it.